dist/src/main/java/io/camunda/application/StandaloneSchemaManager.java

```diff
@@ -100,15 +111,73 @@ public static void main(final String[] args) throws IOException {
-                  "elasticsearch", brokerProperties.getExporters().get("elasticsearch").getArgs())
-              .instantiate(ElasticsearchExporterConfiguration.class);
-      new ElasticsearchExporterSchemaManager(elasticsearchConfig)
-          .createSchema(CURRENT_BROKER_VERSION.toString());
+    try {
+      LOG.info("Creating/updating schema for Camunda using schema manager");
+      getSchemaCreators()
+          .forEach(
+              schemaCreator -> {
+                schemaCreator.createSchema(CURRENT_BROKER_VERSION.toString());
+              });
+      LOG.info("... finished creating/updating schema for Camunda");
+    } catch (final Exception e) {
+      LOG.error("Failed to create/update schema", e);
     }
-    LOG.info("... finished creating/updating Elasticsearch schema for Camunda");
+  }
+
+  private List<SchemaCreator> getSchemaCreators() {
+    final SchemaManagerFactory schemaManagerFactory = getSchemaManagerFactory();
+    return brokerProperties.getExporters().entrySet().stream()
+        .map(entry -> schemaManagerFactory.create(entry.getKey(), entry.getValue()))
+        .toList();
+  }
+
+  private SchemaManagerFactory getSchemaManagerFactory() {
+    final List<String> exporterClassNames =
+        brokerProperties.getExporters().values().stream()
+            .map(ExporterCfg::getClassName)
+            .distinct()
+            .toList();
+
+    if (exporterClassNames.isEmpty()) {
+      throw new IllegalStateException("No exporters configured");
+    }
+    if (exporterClassNames.size() > 1) {
+      throw new IllegalStateException(
+          "Multiple different exporters configured, cannot decide which one to use: "
+              + String.join(", ", exporterClassNames));
+    }
+
+    final Class<?> exporterClass;
+    try {
+      exporterClass = Class.forName(exporterClassNames.getFirst());
+    } catch (final ClassNotFoundException e) {
+      throw new IllegalStateException(
+          "Configured exporter class not found: " + exporterClassNames.getFirst(), e);
+    }
+
+    if (exporterClass.equals(ElasticsearchExporter.class)) {
+      return (id, cfg) -> {
+        final var config =
+            new ExporterConfiguration(id, cfg.getArgs())
+                .instantiate(ElasticsearchExporterConfiguration.class);
+        final var schemaManager = new ElasticsearchExporterSchemaManager(config);
+        return schemaManager::createSchema;
+      };
+    }
+
+    if (exporterClass.equals(OpensearchExporter.class)) {
+      return (id, cfg) -> {
+        final var config =
+            new ExporterConfiguration(id, cfg.getArgs())
+                .instantiate(OpensearchExporterConfiguration.class);
+        final var schemaManager = new OpensearchExporterSchemaManager(config);
+        return schemaManager::createSchema;
+      };
+    }
+
+    throw new IllegalStateException(
+        "Configured exporter is not supported for schema management: "
+            + exporterClassNames.getFirst());
   }
```

Comment:
🔧 I think you can make this factory more static and does not depend on the existing broker configuration, no need to care about the unsupported exporters
```suggestion
    private SchemaManagerFactory getSchemaManagerFactory() {
    return (id, cfg) -> {
      if (ElasticsearchExporter.class.getName().equals(cfg.getClassName())) {
        final var config =
            new ExporterConfiguration(id, cfg.getArgs())
                .instantiate(ElasticsearchExporterConfiguration.class);
        final var schemaManager = new ElasticsearchExporterSchemaManager(config);
        return schemaManager::createSchema;
      } else if (OpensearchExporter.class.getName().equals(cfg.getClassName())) {
        final var config =
            new ExporterConfiguration(id, cfg.getArgs())
                .instantiate(OpensearchExporterConfiguration.class);
        final var schemaManager = new OpensearchExporterSchemaManager(config);
        return schemaManager::createSchema;
      } else {
        return brokerVersion -> {};
      }
    };
  }
```

Reply:
That’s much cleaner! Previously, those checks were tied to constraints like “not supporting multiple backend types in the same run,” while still “supporting multiple backends of the same type,” etc.